src/lib/collectors/utils/requester.ts

```diff
@@ -81,6 +88,12 @@ export class Requester {
 
                     this._redirects.add(newUri, uri);
 
+                    const currentRedirectNumber = this._redirects.calculate(newUri).length;
+
+                    if (currentRedirectNumber > this._maxRedirects) {
+                        reject(`The number of redirects(${currentRedirectNumber}) exceeds the limit(${this._maxRedirects}).`);
```

Comment:
I think you should `return` here otherwise `resolve` or `reject` (again) could get called below.